src/nunit.analyzers/Syntax/ConstraintPartExpression.cs

```diff
@@ -0,0 +1,138 @@
+        {
+            this.expressions = expressions;
+            this.semanticModel = semanticModel;
+        }
+
+        public IEnumerable<ExpressionSyntax> PrefixExpressions
+        {
+            get
+            {
+                // e.g. Has.Property("Prop").Not.EqualTo("1")
+                // -->
+                // Has.Property("Prop"),
+                // Has.Property("Prop").Not
+
+                // Take expressions until found expression returns a constraint
+                return this.expressions
+                    .Where(e => e is MemberAccessExpressionSyntax || e is InvocationExpressionSyntax)
+                    .TakeWhile(e => !ReturnsConstraint(e, this.semanticModel));
+            }
+        }
+
+        public ExpressionSyntax RootExpression
+        {
+            get
+            {
+                return this.expressions.FirstOrDefault(e => ReturnsConstraint(e, this.semanticModel));
+            }
+        }
+
+        public IEnumerable<ExpressionSyntax> SuffixExpressions
+        {
+            get
+            {
+                // e.g. Has.Property("Prop").Not.EqualTo("1").IgnoreCase
+                // -->
+                // Has.Property("Prop").Not.EqualTo("1").IgnoreCase
+
+                // Skip all suffixes, and first expression returning constraint (e.g. 'EqualTo("1")')
+                return this.expressions
+                    .SkipWhile(e => !ReturnsConstraint(e, this.semanticModel))
+                    .Skip(1);
+            }
+        }
+
+        public string[] GetPrefixesNames()
+        {
+            return this.PrefixExpressions
+                .Select(e => GetName(e))
+                .Where(e => e != null)
+                .ToArray();
+        }
+
+        public string[] GetSuffixesNames()
+        {
+            return this.SuffixExpressions
+                .Select(e => GetName(e))
+                .Where(e => e != null)
+                .ToArray();
+        }
+
+        public ExpressionSyntax GetPrefixExpression(string name)
+        {
+            return this.PrefixExpressions.FirstOrDefault(p => GetName(p) == name);
+        }
+
+        public ExpressionSyntax GetSuffixExpression(string name)
+        {
+            return this.SuffixExpressions.FirstOrDefault(s => GetName(s) == name);
+        }
+
+        public string GetConstraintName()
+        {
+            return GetName(this.RootExpression);
+        }
+
+        public ExpressionSyntax GetExpectedArgumentExpression()
+        {
+            if (this.RootExpression is InvocationExpressionSyntax invocationExpression)
+            {
+                return invocationExpression.ArgumentList.Arguments.FirstOrDefault()?.Expression;
+            }
+
+            return null;
+        }
+
+        public IMethodSymbol GetConstraintMethod()
+        {
+            var symbol = this.semanticModel.GetSymbolInfo(this.RootExpression).Symbol;
```

Comment:
This would fail if `this.RootExpression` returns `null`.

Reply:
Fixed